main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"io"
+	"log"
+	"os"
+)
+
+func main() {
+	//TODO move args parsing to a function
+	configFile := flag.String("config", "config.json", "Configuration of the data to be anonymised. Default is 'config.json'")
+	outputFile := flag.String("output", "", "Output file. Default is stdout.")
+	flag.Parse()
+	log.Printf("Using configuration in file %s\n", *configFile)
+	r := initReader(flag.Arg(0))
+	w := initWriter(*outputFile)
+	//TODO create the anons array using the config provided
+	anons := []anonymisation{hash, identity, identity, outcode}
+	i := 0
+
+	for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		} else if pe, ok := err.(*csv.ParseError); ok && pe.Err == csv.ErrFieldCount {
+			// we just print the error and skip the column
```

Comment:
I think this behaviour could potentially be dangerous - imagine if the malformed row still had information in it that should be sanitised but we skipped it because of a single comma out of place). I think that we should, until we have a way to configure this behaviour, have the safest defaults we can think of which in this case would be to bail out of the entire file. Later on, we can make skipping parse errors a setting you can opt-in to.

Reply:
Oh duh, the behaviour is actually to skip the entire line. In that case, it seems safe albeit with the risk of losing data out the end.

Reply:
Just created this: https://github.com/intenthq/anon/issues/2